ibis/sql/postgres/compiler.py

```diff
@@ -294,9 +304,24 @@ def _strftime(t, expr):
     return reduce(sa.sql.ColumnElement.concat, reduced)
 
 
-def _distinct_from(a, b):
-    return (((a != None) | (b != None)) &  # noqa
-            sa.func.coalesce(a != b, True))
+class array_search(expression.FunctionElement):
+    type = sa.INTEGER()
+    name = 'array_search'
+
+
+@compiles(array_search)
+def postgresql_array_search(element, compiler, **kw):
```

Comment:
Slightly less cryptic hack for finding an element in a postgres array.

Reply:
Later versions of postgres have a function to do this (I think starting in 9.5 or 9.6), and there isn't really a way to get the client or server version without executing a query so I wrote this to work on all versions.

We could can this monster by not supporting older versions of postgres, but I don't think this one function is really enough reason to abandon all older versions of postgres.

We should probably just follow whatever the postgres people do regarding supporting particular versions. I think 9.3 is the oldest version they support ATM.

Reply:
BTW I've introduced a `client.version` property for each backend in the upcoming PR, it's alchemy `dialect.server_version_info` in case of postgres which if queried once per connection.